flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterGateway.java

```diff
@@ -227,5 +229,19 @@ void failSlot(final ResourceID taskManagerId,
 	 */
 	void heartbeatFromResourceManager(final ResourceID resourceID);
 
+	/**
+	 * Request the details of the executed job.
+	 *
+	 * @param timeout for the rpc call
+	 * @return Future details of the executed job
+	 */
 	CompletableFuture<JobDetails> requestJobDetails(@RpcTimeout Time timeout);
```

Comment:
are we still keeping this for the job master?

Reply:
`JobDetails` are independent of the `ExecutionGraphCache` and still needed. I simply added java docs.